### ISSUE TITLE
OpenCV 3.0 install error linking libX11.so.6 opengl_interop.cpp.o

### ISSUE THREAD

**Issue by opencv-pushbot** [Itseez/opencv#5067]
Transferred from http://code.opencv.org/issues/4477
```
|| m ho on 2015-07-09 15:02
|| Priority: Normal
|| Affected: branch 'master' (3.0-dev)
|| Category: None
|| Tracker: Bug
|| Difficulty: 
|| PR: 
|| Platform: x86 / Linux
```

OpenCV 3.0 install error linking libX11.so.6 opengl_interop.cpp.o
-----------
```
I installed OpenCV 3.0 successfully 2 days ago and realized I forgot to add the WITH_OPENGL=ON in the cmake call.  I made the change and when I tried to build it I get the following errors:

@
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D INSTALL_C_EXAMPLES=ON -D INSTALL_PYTHON_EXAMPLES=ON -D OPENCV_EXTRA_MODULES_PATH=~/opencv_contrib/modules -D BUILD_EXAMPLES=ON ..

[ 36%] Building CXX object modules/ts/CMakeFiles/opencv_ts.dir/src/ocl_perf.cpp.o
Linking CXX static library ../../lib/libopencv_ts.a
[ 36%] Built target opencv_ts
[ 38%] Building CXX object apps/annotation/CMakeFiles/opencv_annotation.dir/opencv_annotation.cpp.o
[ 38%] Building CXX object modules/features2d/CMakeFiles/opencv_features2d.dir/src/orb.cpp.o
Linking CXX executable ../../bin/opencv_annotation
[ 38%] Built target opencv_annotation
[ 38%] Building CXX object samples/opengl/CMakeFiles/example_opengl_opengl_interop.dir/opengl_interop.cpp.o
ptxas /tmp/tmpxft_000047e7_00000000-20_split_merge.compute_11.ptx, line 873; warning : Double is not supported. Demoting to float
[ 38%] Building CXX object modules/features2d/CMakeFiles/opencv_features2d.dir/src/blobdetector.cpp.o
Linking CXX executable ../../bin/opengl-example-opengl_interop
/usr/bin/ld: CMakeFiles/example_opengl_opengl_interop.dir/opengl_interop.cpp.o: undefined reference to symbol 'XGetWindowAttributes'
//usr/lib/i386-linux-gnu/libX11.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [bin/opengl-example-opengl_interop] Error 1
make[1]: *** [samples/opengl/CMakeFiles/example_opengl_opengl_interop.dir/all] Error 2
@

So I did the most common thing and tried to cmake without the OpenGL, and I still get the same error while trying to build.
```

History
-------
##### Alexander Alekhin on 2015-07-09 16:00
```
Changing CMake parameters "on the fly" is not stable and can provide wrong unexpected results. Please remove build directory before CMake relaunch (build from scratch).

I tried these CMake flags and build is OK:
-DWITH_OPENGL=ON -DWITH_QT=ON -DBUILD_EXAMPLES=ON

Currently I can't check build with GTK UI backend (I have GTK3 and there is no support for OpenGL in this configuration)

Also you can try to add this CMake parameter (I can't check it myself, because I can't reproduce this problem):
-DOPENCV_LINKER_LIBS="X11"
But I'm not sure that it will work, because it was created for OpenCV modules, not for samples.
```
##### m ho on 2015-07-10 01:44
```
Thanks... Your right, I should have cleared out the build folder; however still got the same error (clearing out the build folder did let me getting it working without OpenGL.)  I will try changing the CMake parameter and see if that will get me running with OpenGL. Thanks!
```

**Comment by TheodoreT** [comment]
Same compilation problem here. However, adding the CMake parameter that Alexander proposed solved the issue.

**Comment by mshabunin** [comment]
Issue should be fixed by #5168
